packages/auth/src/const.ts

```diff
@@ -3,3 +3,13 @@ export const REFRESH_TOKEN = "refresh_token";
 export const ID_TOKEN = "id_token";
 
 export const POST_LOGOUT_URI_KEY = "post_logout_uri";
+
+// These come from api/config/rolepermission.php
+export enum RoleName {
+  Guest = "guest",
+  BaseUser = "base_user",
+  Applicant = "applicant",
+  PoolOperator = "pool_operator",
+  RequestResponder = "request_responder",
+  PlatformAdmin = "platform_admin",
+}
```

Comment:
Two things for this:

1. If we export this from the `auth/src/index.tsx` file, we won't need the `src/cont` part of the import path.
2. More philosophical but, [Typescript enums suck](https://www.typescriptlang.org/docs/handbook/enums.html#const-enum-pitfalls) so if we can avoid them, that would be great

## Enums

I know we use enums already but they are almost exclusively generated and out of our control. However, when possible, it is usually best to [avoid typescript enums](https://www.executeprogram.com/blog/typescript-features-to-avoid#:~:text=I.-,Avoid%20enums).

### Alternative

Ref: https://www.niraj.life/blog/avoid-using-enums-in-typescript#:~:text=Unions%20are%20the%20easiest%20way.%20But%20you%20can%20also%20try%20objects%3A

```tsx
const ROLE_NAME = {
  Guest: 'guest',
} as const;

type ObjectValues<T> = T[keyof T];
type RoleName = ObjectValues<typeof ROLE_NAME>;
```

Reply:
Haha, I first wrote this as an object - I've heard the internet's enum-hate for typescript.  :laughing: I ended up going with the familiarity of the enum but you've convinced me to switch it back.
ref: [switch role name enum to object](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5852/commits/2d4e6a702c764b68f199129986c126d8725ab077)

Good catch on the imports.
ref: [update imports to index.tsx](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5852/commits/abd72defd32afdf5603fb0b65c15883953ec85a7)